assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -142,10 +146,39 @@ const PageEdit = ( {
 		}
 	}, [ mediaType, mediaUrl ] );
 
+	// If there is more than one immovable block, leave only the last and remove the others.
+	useEffect( () => {
+		if ( immovableBlocks.length > 1 ) {
+			immovableBlocks.pop();
```

Comment:
I'm not sure mutating it is such a great idea given that it's used as a dependency for this and other hooks. I think all hook dependencies ought to be kept immutable.

Rather, just `slice` it in the next line:

```js
// keep first
const blocksToRemove = immovableBlocks.slice( 1 ).map( ( { clientId: blockId } ) => blockId );
// keep last
const blocksToRemove = immovableBlocks.slice( 0, -1 ).map( ( { clientId: blockId } ) => blockId );
```

Reply:
Fixing in a minute.